### ISSUE TITLE
"yield" keyword with no value causes an error

### ISSUE THREAD

**Issue by hegemonic** [eslint/espree#121]
According to the [ES 2015 spec](https://people.mozilla.org/~jorendorff/es6-draft.html#sec-generator-function-definitions), a `YieldExpression` with no value is valid. (You can see some examples in [Axel Rauschmayer's article on generators](http://www.2ality.com/2015/03/no-promises.html).) However, Espree currently throws an error when it attempts to parse the following code, with generators enabled:

```js
function* asyncFunc() {
    var caller = yield;
}
```

Changing `yield` to `yield 1` allows parsing to succeed.

**Comment by nzakas** [comment]
Thanks, I'll take a look.

**Comment by nzakas** [comment]
Working on this.
